api/src/services/config_service.rs

```diff
@@ -221,14 +298,40 @@ impl TryFrom<&Companion> for ServiceConfig {
     }
 }
 
+impl From<&CompanionType> for ContainerType {
+    fn from(t: &CompanionType) -> Self {
+        match t {
+            CompanionType::Application => ContainerType::ApplicationCompanion,
+            CompanionType::Service => ContainerType::ServiceCompanion,
+        }
+    }
+}
+
 #[cfg(test)]
 mod tests {
     use super::*;
+    use sha2::{Digest, Sha256};
+
+    macro_rules! service_config {
+        ( $name:expr ) => {{
+            let mut hasher = Sha256::new();
+            hasher.input($name);
+            let img_hash = &format!("sha256:{:x}", hasher.result_reset());
+
+            ServiceConfig::new(String::from($name), Image::from_str(&img_hash).unwrap())
+        }};
+    }
+
+    macro_rules! config_from_str {
```

Comment:
I didn't know creating macros would be so simple :+1:

Reply:
Rust is just awesome. :wink: